x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/TransportSubmitAsyncSearchAction.java

```diff
@@ -83,7 +83,10 @@ public void onResponse(AsyncSearchResponse searchResponse) {
                                 onFatalFailure(searchTask, cause, false, submitListener);
                             } else {
                                 final String docId = searchTask.getSearchId().getDocId();
-                                store.storeInitialResponse(docId, searchTask.getOriginHeaders(), searchResponse,
+                                // creates the fallback response if the node crashes/restarts in the middle of the request
+                                // TODO: store intermediate results ?
```

Comment:
can you elaborate on this TODO? does it revolve around resiliency?

Reply:
It does, yes that's one of the follow up question we have in the meta issue.

Reply:
I thought so, I wonder if we need the TODO in the code then, cause we are tracking this anyways elsewhere.